app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -142,6 +142,16 @@ def pre_prov
         page << javascript_for_miq_button_visibility(session[:changed])
         @edit[:src_vm_id] = params[:sel_id].to_i
       end
+    elsif params[:hide_deprecated_templates]
+      @edit = session[:edit]
+      @edit[:hide_deprecated_templates] = params[:hide_deprecated_templates] == "true"
+      templates = Rbac.filtered(@edit[:template_kls].eligible_for_provisioning).sort_by { |a| a.name.downcase }
+      build_vm_grid(templates, @edit[:vm_sortdir], @edit[:vm_sortcol], build_template_filter)
```

Comment:
You're totally right, it's copied/pasted. I'll give a shot to consolidating them.